nodejs/discord-bots/discordjs-logbot/index.js

```diff
@@ -0,0 +1,203 @@
+    // get user tag
+    const tag = newMessage.author.tag;
+    // make a date variable
+    var date = new Date();
+    // send date to the console
+    // get the servers id
+    var serverid = newMessage.guild.id;
+    // if the message is not from the bot, add the message to messages.txt
+    fs.appendFile("./logs/" + newMessage.channel.id + '.txt', tag + ': ' + oldMessage.content + ' EDITED TO ' + newMessage.content + '\n', (err) => {
+        if (err) throw err;
+    });
+});
+// check if a file is over 10 mb, if it is, delete it
+function checkFileSize(file) {
+    // get the file size
+    var stats = fs.statSync(file);
+    // get the file size in bytes
+    var fileSizeInBytes = stats["size"];
+    // convert the file size to megabytes
+    var fileSizeInMegabytes = fileSizeInBytes / 1000000.0;
+    // if the file is over 10 mb, delete it
+    if (fileSizeInMegabytes > 10) {
+        fs.unlink(file, (err) => {
+            if (err) throw err;
+        });
+    }
+}
+// check if file exists
+function checkFileExists(file) {
+    // if the file exists, return true
+    if (fs.existsSync(file)) {
+        return true;
+    }
+    // if the file does not exist, return false
+    else {
+        return false;
+    }
+}
+// display message when someone sends a message
+client.on('message', message => {
+    // if file does not exist
+    if (!checkFileExists("./logs/" + message.channel.id + '.txt')) {
+        fs.writeFileSync("./logs/" + message.channel.id + ".txt", JSON.stringify("", null, 4));
+    }
+    checkFileSize("./logs/" + message.channel.id + '.txt');
+    // get user tag
+    const tag = message.author.tag;
+    // make a date variable
+    var date = new Date();
+    // send date to the console
+    // get the servers id
+    var serverid = message.channel.id;
+    // if the message is not from the bot, add the message to messages.txt
+    fs.appendFile("./logs/" + serverid + '.txt', tag + ': ' + message.content + '\n', (err) => {
+        if (err) throw err;
+    });
+    if(message.author.bot) return;
+    // set the prefix to the prefix in config.json
+    var prefix = config.prefix;
+    // if the message is "stop" and the message authors id is 545636523580850186 then stop the bot
+    if (message.content == prefix + 'stop' && message.author.id == config.ownerid) {
+        message.reply('Stopping...');
+        client.destroy();
+        process.exit();
+    }
+    // if the message is prefix + "download", send ./logs/guildname.txt to hastebin
+    if (message.content == prefix + 'download' && message.member.id == message.guild.ownerId) {
+        // store the last 100 lines of the file
+        var lines = fs.readFileSync("./logs/" + message.guild.id + '.txt').toString().split('\n').slice(-501);
+        // replace every "ॐ" with a newline
+        var logs = lines.join('\n');
+        // send the archive to hastebin
+        fetch('http://132.145.29.171:1035/documents', {
```

Comment:
What is this URL?

Reply:
a hastebin install, probably will change it in a second to be the web server that the bot makes

Reply:
Use https://paste.sparked.host please

Reply:
fixed